### DISCUSSION TITLE
How to set an IP address to vm

### DISCUSSION THREAD

**Discussion by thakkarjigneshX** [vmware/govmomi#2930] [Community Help] [answered]
Hello,
I created the datacenter, cluster, datastore and vm using below commands.
However, when I check the vm ip, it shows empty. How can I populate the same? What am I missing?

```
govc datacenter.create TestDC
govc cluster.create -dc=TestDC TestCluster
govc cluster.add -dc=TestDC -hostname x.x.x.x -username user -password pass -noverify
govc datastore.create -dc=TestDC -type local -name localDatastore111 -path /var/local x.x.x.x
govc vm.create -c 2 -m 10000 -disk 200GB -net.adapter e1000 -disk.controller pvscsi -dc=TestDC TestVM 
govc vm.info -dc=TestDC TestVM
Name:           TestVM
  Path:         /TestDC/vm/TestVM
  UUID:         88aeeca4-87a6-5b24-8e05-fac539959a2e
  Guest name:   otherGuest
  Memory:       10000MB
  CPU:          2 vCPU(s)
  Power state:  poweredOn
  Boot time:    2022-08-15 04:59:50.317078979 -0700 PDT
  IP address:
  Host:         10.46.213.108
nutanix@NTNX-diagnostics:~$

```
Also, govc metric.ls -dc=TestDC TestVM shows the list of available metrics.
Are these metrics getting auto-populated or we can populate a few/some of them for a given vm?

**Comment by tenthirtyam** [comment]
See [vm.create](https://github.com/vmware/govmomi/blob/master/govc/USAGE.md#vmcreate) and [vm.customize](https://github.com/vmware/govmomi/blob/master/govc/USAGE.md#vmcustomize).

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Reply by villiwalla** [reply, marked as answer]
```bash
./govc vm.customize \ 
-k=true -u='*********' \
-dc='vSAN Datacenter' \
-vm='DEV-UBNT-TMP01' \
-name='dev-ubnt-template' \
-mac=00:50:56:94:ae:80 \
-ip=192.168.50.77 \ 
-netmask=255.255.255.0 \
-ip6 '2001:db8::1/64' \
-domain='ubnt01.infra.local' \
-vim-namespace=6.7
```

to assign an IP address to the only network adapter. I need to set a fixed IP address even though there is a DHCP server in the network. In the virtual machine tasks, the task executes successfully, but it changes the network interface's autoconnect setting to false. Is it possible to assign static IP addresses?
![image](https://github.com/user-attachments/assets/1ea355ae-f936-4b50-ae43-a5c95176bf2e)

> Reconfigured DEV-UBNT-TMP01 on node-2.dclogic.ru in vSAN Datacenter.  
>  
> Modified:  
>  
> config.tools.pendingCustomization: <unset> -> &quot;/vmfs/volumes/vsan:521657d88039b1e6-886031b60c7873df/30521267-4405-29d3-adcb-e4434b5c70b0/imcf-j5fvc6&quot;; 
> 
>  Added:  
>  
> config.extraConfig(&quot;tools.deployPkg.fileName&quot;): (key = &quot;tools.deployPkg.fileName&quot;, value = &quot;imcf-j5fvc6&quot;); 
> 
>  Deleted:  

or use:

```bash
./govc vm.change \
-k=true \
-u='****' \
-dc='vSAN Datacenter' \
-vim-version=6.7.0 \
-json \
-vm DEV-UBNT-TMP01 \
-e SET.guest.ipAddress=192.168.50.72 \
-e SET.guest.hostName=clone02.infrabase.local
```

task info:

> Modified: 
> 
>  Added: 
> 
> config.extraConfig("SET.guest.ipAddress"): (key = "SET.guest.ipAddress", value = "192.168.50.72"); 
> 
> config.extraConfig("SET.guest.hostName"): (key = "SET.guest.hostName", value = "clone02.infrabase.local"); 
> 
>  Deleted: 

with no effect on the virtual machine

env:

```json
{
	"about": {
		"name": "VMware vCenter Server",
		"fullName": "VMware vCenter Server 6.7.0 build-11727113",
		"vendor": "VMware, Inc.",
		"version": "6.7.0",
		"build": "11727113",
		"localeVersion": "INTL",
		"localeBuild": "000",
		"osType": "linux-x64",
		"productLineId": "vpx",
		"apiType": "VirtualCenter",
		"apiVersion": "6.7.1",
		"instanceUuid": "ffaae73b-bb22-4c7d-9b1b-1e783aa0c466",
		"licenseProductName": "VMware VirtualCenter Server",
		"licenseProductVersion": "6.0"
	}
}
```
